---
title: "00-data_preprocessing"
output: html_document
date: "2023-07-05"
---

Credit for the code in this notebook: https://www.costalab.org/wp-content/uploads/2022/11/handout_day5.html#3_RNASeq_Normalization
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("SummarizedExperiment")
BiocManager::install("EDASeq")
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
library(tidyverse)
library(TCGAbiolinks)
library(SummarizedExperiment)
library(EDASeq)
```

```{r}
query = GDCquery(
  project = "TCGA-LIHC",
  data.category = "Transcriptome Profiling", 
  data.type = "Gene Expression Quantification",
  experimental.strategy = "RNA-Seq",
  workflow.type = "STAR - Counts",
  sample.type = c("Primary Tumor", "Solid Tissue Normal"))

GDCdownload(query)
```

```{r}
lihc_res = getResults(query)
colnames(lihc_res)
```

```{r}
table(lihc_res$sample_type)
```

```{r}
tcga_data = GDCprepare(query)
dim(tcga_data)
```

```{r}
colnames(colData(tcga_data))
```

```{r}
table(tcga_data@colData$definition)
```

```{r}
table(tcga_data@colData$ajcc_pathologic_stage)
```

```{r}
dim(assay(tcga_data))    
```

```{r}
head(assay(tcga_data)[,1:10]) # first 6 genes of the first 10 samples
```

```{r}
saveRDS(object = dataFilt,
        file = "dataFilt.RDS",
        compress = FALSE)

```

```{r}
tcga_data = readRDS(file = "tcga_data.RDS")
```

```{r}
genes<- as.data.frame(rowData(tcga_data))
```


```{r}
dataPrep <- TCGAanalyze_Preprocessing(
  tcga_data,
  cor.cut = 0.6,
  filename = 'aaic_plot.png',
  width = 1000,
  height = 1000,
  )
```

```{r}
dataNorm <- TCGAbiolinks::TCGAanalyze_Normalization(dataPrep, geneInfoHT, method = "gcContent")
dataNorm <- na.omit(dataNorm)
```

```{r}
dataFilt <- TCGAbiolinks:::TCGAanalyze_Filtering(dataNorm, method = "quantile", qnt.cut = 0.25)
```

```{r}
tgca_data
```


